src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -83,6 +83,7 @@ public class DefaultModelMetadata : ModelMetadata
             _detailsProvider = detailsProvider;
             _details = details;
             _modelBindingMessageProvider = modelBindingMessageProvider;
+            ContainerMetadata = details.ContainerMetadata;
```

Comment:
Prefer not to use `details` this way in the constructor. We don't really need to store a separate value at all.

Suggest going back to a `virtual` property in `ModelMetadata` (that returns `null`) and having the `override` here `return _details.ContainerMetadata`. But it's certainly possible I can be convinced otherwise.